lib/Sema/CSSimplify.cpp

```diff
@@ -1352,16 +1352,39 @@ ConstraintSystem::matchTupleTypes(TupleType *tuple1, TupleType *tuple2,
 // corresponding function type representations and the given match kind.
 static bool matchFunctionRepresentations(FunctionTypeRepresentation rep1,
                                          FunctionTypeRepresentation rep2,
-                                         ConstraintKind kind) {
+                                         ConstraintKind kind,
+                                         ConstraintLocatorBuilder locator) {
   switch (kind) {
   case ConstraintKind::Bind:
   case ConstraintKind::BindParam:
   case ConstraintKind::BindToPointerType:
   case ConstraintKind::Equal:
     return rep1 != rep2;
+      
+  case ConstraintKind::Subtype: {
+    auto last = locator.last();
```

Comment:
Maybe we can factor out a function or method somewhere to decide if a conversion is valid

Reply:
Humm, thanks :)

> Maybe we can factor out a function or method somewhere to decide if a conversion is valid

I was thinking maybe a `matchFunctionRepresentationConversion` as a static method here in `ConstraintSystem`? What do you think? @slavapestov @xedin

Reply:
Is on the last commit, let me know what you guys think :)